### PR TITLE
init commit

### PR DESCRIPTION
**Pull request by entertainment-veks**

### CODE REVIEW

File: src/main/java/hibernate/model/ShoppingCart.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import javax.persistence.Id;
+import javax.persistence.MapsId;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+
+@Entity(name = "shopping_cart")
+public class ShoppingCart {
+    @Id
+    private Long id;
+    @OneToMany(fetch = FetchType.EAGER)
```
Comment (Artificial-Friend) [inline src/main/java/hibernate/model/ShoppingCart.java:16]: It's not wrong but at today's lesson we talked that it's not a good practice and FETCH joining is better.

**Review by Artificial-Friend — APPROVED** [review]

File: src/main/java/hibernate/dao/impl/ShoppingCartDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query = session.createQuery("from shopping_cart"
+                            + " WHERE user.id = :userId", ShoppingCart.class);
+            query.setParameter("userId", user.getId());
+            return query.getSingleResult();
```
Comment (NikolayFatkullin) [inline src/main/java/hibernate/dao/impl/ShoppingCartDaoImpl.java:44]:
```suggestion
            return session.createQuery("from shopping_cart"
                            + " WHERE user.id = :userId", ShoppingCart.class)
                      .setParameter("userId", user.getId())
                      .getSingleResult();
```

**Review by NikolayFatkullin — APPROVED** [review]

File: src/main/java/hibernate/Main.java [hunk 3]
```diff
@@ -60,6 +60,21 @@ public static void main(String[] args) {
...
+        ShoppingCartService shoppingCartService =
+                (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+        shoppingCartService.addSession(movieSessionLotr, bob);
+
+        System.out.println(shoppingCartService.getByUser(bob));
+
+        shoppingCartService.clear(shoppingCartService.getByUser(bob));
+
+        System.out.println(shoppingCartService.getByUser(bob));
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/Main.java:78]: why so many empty lines?

File: src/main/java/hibernate/dao/impl/ShoppingCartDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.update(shoppingCart);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert Shopping cart entity "
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/dao/impl/ShoppingCartDaoImpl.java:63]:
```suggestion
            throw new DataProcessingException("Can't update Shopping cart entity "
```

File: src/main/java/hibernate/model/ShoppingCart.java (hunk 1 above)
Comment (KseniiaMakarova) [inline src/main/java/hibernate/model/ShoppingCart.java:16]: I agree - we should never eager-fetch in ManyToMany and OneToMany relationships

File: src/main/java/hibernate/model/ShoppingCart.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    private List<Ticket> tickets;
+    @OneToOne
+    @MapsId
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/model/ShoppingCart.java:19]: let's make this column be named `id`

File: src/main/java/hibernate/service/impl/AuthentificationServiceImpl.java [hunk 6]
```diff
@@ -14,6 +15,9 @@
     @Inject
     private UserService userService;
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/service/impl/AuthentificationServiceImpl.java:17]: redundant empty line

File: src/main/java/hibernate/service/impl/ShoppingCartServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import hibernate.model.ShoppingCart;
+import hibernate.model.Ticket;
+import hibernate.model.User;
+import hibernate.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/service/impl/ShoppingCartServiceImpl.java:17]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
let's also look at the repository settings and make PRs be checked by travis

File: src/main/java/hibernate/dao/impl/ShoppingCartDaoImpl.java [hunk 8]
```diff
@@ -38,8 +38,9 @@ public ShoppingCart add(ShoppingCart shoppingCart) {
     @Override
     public ShoppingCart getByUser(User user) {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-            return session.createQuery("from shopping_cart"
-                    + " WHERE user.id = :userId", ShoppingCart.class)
+            return session.createQuery("from shopping_cart s"
+                    + " JOIN ticket t"
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/dao/impl/ShoppingCartDaoImpl.java:42]: let's use `join fetch`, also - we should be able to get shopping cart even if it is empty - is it possible now?

File: src/main/java/hibernate/model/ShoppingCart.java [hunk 9]
```diff
@@ -2,21 +2,17 @@
 
 import java.util.List;
 import java.util.Objects;
-import javax.persistence.Entity;
-import javax.persistence.FetchType;
-import javax.persistence.Id;
-import javax.persistence.MapsId;
-import javax.persistence.OneToMany;
-import javax.persistence.OneToOne;
+import javax.persistence.*;
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/model/ShoppingCart.java:5]: I believe the checkstyle doesn't pass now - let's configure travis properly

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
the code is ok, could you please share the screenshots in Slack

- of `mvn clean package` run locally
- of repository settings that enable integration with Travis

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**